src/new/wallet.js

```diff
@@ -0,0 +1,441 @@
+  }
+
+  getBalance(tokenUid) {
+    if (!tokenUid && !this.token) {
+      throw Error('Token must be set in the class instance or passed as parameter in the method.')
+    }
+    const uid = tokenUid || this.token.uid;
+    const historyTransactions = this.getTxHistory();
+    return wallet.calculateBalance(Object.values(historyTransactions), uid);
+  }
+
+  getTxHistory() {
+    const data = wallet.getWalletData();
+    const historyTransactions = 'historyTransactions' in data ? data['historyTransactions'] : {};
+    return historyTransactions;
+  }
+
+  setState(state) {
+    this.state = state;
+    this.emit('state', state);
+  }
+
+  onNewTx(wsData) {
+    const newTx = wsData.history;
+
+    // TODO we also have to update some wallet lib data? Lib should do it by itself
+    const walletData = wallet.getWalletData();
+    const historyTransactions = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
+    const allTokens = 'allTokens' in walletData ? walletData.allTokens : [];
+
+    let isNewTx = true;
+    if (newTx.tx_id in historyTransactions) {
+      isNewTx = false;
+    }
+
+    wallet.updateHistoryData(historyTransactions, allTokens, [newTx], null, walletData);
+
+    if (isNewTx) {
+      this.emit('new-tx', newTx);
+    } else {
+      this.emit('update-tx', newTx);
+    }
+    return;
+
+    // TODO XXX Uncomment the following block to keep track of the balance.
+    /*
+    // We need to reload it because it was modified by updateHistoryData.
+    const newWalletData = wallet.getWalletData();
+    const { keys } = newWalletData;
+
+    const updatedHistoryMap = {};
+    const updatedBalanceMap = {};
+    const balances = this.getTxBalance(newTx);
+
+    // we now loop through all tokens present in the new tx to get the new history and balance
+    for (const [tokenUid, tokenTxBalance] of Object.entries(balances)) {
+      // we may not have this token yet, so state.tokensHistory[tokenUid] would return undefined
+      const currentHistory = state.tokensHistory[tokenUid] || [];
+      const newTokenHistory = addTxToSortedList(tokenUid, tx, tokenTxBalance, currentHistory);
+      updatedHistoryMap[tokenUid] = newTokenHistory;
+      // totalBalance should not be confused with tokenTxBalance. The latter is the balance of the new
+      // tx, while the former is the total balance of the token, considering all tx history
+      const totalBalance = getBalance(tokenUid);
+      updatedBalanceMap[tokenUid] = totalBalance;
+    }
+    const newTokensHistory = Object.assign({}, state.tokensHistory, updatedHistoryMap);
+    const newTokensBalance = Object.assign({}, state.tokensBalance, updatedBalanceMap);
+
+    storage.setItem('local:tokens:history', newTokensHistory);
+    storage.setItem('local:tokens:balance', newTokensBalance);
+    */
+  };
+
+  /**
+   * Send a transaction with multi tokens
+   * @param {Object} data Array of {'address', 'value', 'token'}
+   */
+  sendMultiTokenTransaction(data) {
```

Comment:
This name seems misleading. I think the main thing is that is sends multiple outputs, not necessarily multiple tokens. I'd suggest naming this `sendTransaction`. As for the other function already named like this, I suggested refactoring it. We can:
- change name to `sendOne` or something like that;  
- remove it all together. People would use this function with an array with only one item.

Reply:
I agree.

Reply:
I agree, it has this problem and others already written on the code. This method deserves a refactor to be used. Created an issue to do it later. https://github.com/HathorNetwork/hathor-wallet-lib/issues/51